lib/dash/dash_parser.js

```diff
@@ -1973,6 +1973,24 @@ shaka.dash.DashParser = class {
 
     // Parse Representations into Streams.
     const representations = TXml.findChildren(elem, 'Representation');
+    if (!this.config_.dash.ignoreSupplementalCodecs) {
+      // Duplicate representations with their supplementalCodecs
+      let supplementalRepresentations = representations.filter((rep) => {
```

Comment:
I feel like the multiple stages of `filter`, `map`, then `append` are obfuscating what this process does. When I first read this I thought that the comment above applied only to the `filter` operation and got confused about how it applied.

So in this case, I think that doing this as a `for...of` would be shorter, more readable, and probably more efficient.
Something like (not tested):

```js
for (const rep of representations) {
  const supplementalCodecs = TXml.getAttributeNS(
      rep, shaka.dash.DashParser.SCTE214_, 'supplementalCodecs');
  if (supplementalCodecs) {
    const obj = shaka.util.ObjectUtils.cloneObject(rep);
    obj.attributes['codecs'] = supplementalCodecs.split(' ').join(',');
    representations.push(obj);
  }
}
```

(This could also be done as a `reduce`, if we want to stick to functional methods).

Reply:
Done!